packages/block-library/src/image/deprecated.js

```diff
@@ -539,4 +543,95 @@ const v5 = {
 	},
 };
 
-export default [ v5, v4, v3, v2, v1 ];
+/**
+ * Deprecation for adding width and height as style rules on the inner img.
+ * It also updates the widht and height attributes to be strings instead of numbers.
```

Comment:
It looks like the width and height are still numbers here.

Reply:
I think the PR description remained behind on the actual end result of what's implemented. The updating to string never happened and this comment is wrong.

Reply:
> The updating to string never happened and this comment is wrong.

Do you know if that is that stil true for WordPress 6.3.2? We're seeing a lot of 'block recovery' in the editor for images like these:

```html
<!-- wp:image {"id":1,"width":200,"height":100,"sizeSlug":"large","linkDestination":"none"} -->
<figure class="wp-block-image size-large is-resized"><img src="https://example.com/wp-content/uploads/2023/10/example.svg" alt="" class="wp-image-1" style="width:200px;height:100px"/></figure>
<!-- /wp:image -->
```

Updating the `width` and `height` attributes from numbers to strings (`"200px"` and `"100px"`) seems to fix the blocks.

Reply:
> Do you know if that is that still true for WordPress 6.3.2?

Currently, they should be getting migrated to strings. However, it will only migrate for valid markup.

> We're seeing a lot of 'block recovery' in the editor for images like these:

Yeah, that markup looks invalid. If the block attributes are correct, the `img`'s `width` and `height` attributes should be there. Or if the markup is correct, the block attributes should be strings with `px` units.

Do you happen to know if that markup was generated with gutenberg trunk (at some point), a proper gutenberg plugin release, or a full WordPress release? If it was generated with some release code, we probably need to update some deprecations—we shouldn't be getting block recovery errors from release code.

> Updating the width and height attributes from numbers to strings ("200px" and "100px") seems to fix the blocks.

Yep, this is what I was going to suggest. It's the best workaround for now.

Reply:
> If it was generated with some release code, we probably need to update some deprecations—we shouldn't be getting block recovery errors from release code.

That is what I was thinking too.

> Do you happen to know if that markup was generated with gutenberg trunk (at some point), a proper gutenberg plugin release, or a full WordPress release?

Not Gutenberg trunk. I assumed a full WordPress release, but I now discovered that the Gutenberg plugin is also installed (inactive, so I'm not sure if it has been active in the past and with which version).

We've updated all posts in the meantime, which has resolved the issue for our client.

Reply:
#56916 was submitted, probably related to this PR.

Upgrading from 6.2 to 6.3 or 6.4 seems to break the Image block. This might be related to the fact that in 6.2, the `width` and `height` attribute values are number types.